examples/sentencepiece/train.py

```diff
@@ -0,0 +1,162 @@
+import torch
+import os
+import logging
+import argparse
+
+r"""
+This example is similar to the one in examples/text_classification.
+The only difference is the dataset, which applies SentencePiece as encoder.
+The subword method in SentencePiece was tested based on YelpReviewFull and
+we are able to reproduce the results from fastText.
+"""
+
+
+def generate_batch(batch):
+    label = torch.tensor([entry[0] for entry in batch])
+    text = [entry[1] for entry in batch]
+    offsets = [0] + [len(entry) for entry in text]
+
+    offsets = torch.tensor(offsets[:-1]).cumsum(dim=0)
+    text = torch.cat(text)
+    return text, offsets, label
+
+
+from torch.utils.data import DataLoader
```

Comment:
Imports should always be at the top of the file. Install a formatter such as https://github.com/hhatto/autopep8 - you can then apply it via something like https://github.com/google/vim-codefmt

Reply:
I personally like [black](https://pypi.org/project/flake8-black/), since it autoformats the code for you, and [isort](https://github.com/timothycrosley/isort).